modules/gapi/CMakeLists.txt

```diff
@@ -383,10 +385,12 @@ if(HAVE_ONNX)
   endif()
 endif()
 
+add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/3rdparty/vasot ${CMAKE_CURRENT_BINARY_DIR}/3rdparty/vasot)
```

Comment:
Why
```
${CMAKE_CURRENT_BINARY_DIR}/3rdparty/vasot
```
needs to be added?

Reply:
One IOS build is failing: https://github.com/opencv/opencv/actions/runs/6937152923/job/18870654308?pr=24224#step:7:297

```
CMake Error at modules/gapi/CMakeLists.txt:388 (add_subdirectory):
  add_subdirectory not given a binary directory but the given source
  directory
  "/Users/opencv-cn/GHA-OCV-1/_work/opencv/opencv/opencv/modules/gapi/3rdparty/vasot"
  is not a subdirectory of
  "/Users/opencv-cn/GHA-OCV-1/_work/opencv/opencv/opencv/modules/world".
  When specifying an out-of-tree source a binary directory must be explicitly
  specified.
Call Stack (most recent call first):
  modules/world/CMakeLists.txt:13 (include)
  modules/world/CMakeLists.txt:32 (include_one_module)
```

Reply:
Basically it means the static build may be broken because of that.. but what makes OT different from any other source-level subdirectory in this case?

Reply:
Thanks a lot, fixed!